src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -100,6 +106,18 @@ public static void Initialize()
 
             if (TracerSettings.LoadTracerAtStartup)
             {
+                // Setup the instrumentations that have additional setup occurring during AssemblyLoad
+                // -> this should be refactored in a separate PR
+                // e.g. we could have a static method that returns a collection of initializers
+                //      and TracerSettings.EnabledInstrumentations would be passed as input
+#if NETCOREAPP3_1_OR_GREATER
+
+                if (TracerSettings.EnabledInstrumentations.Contains(TracerInstrumentation.MySqlData))
+                {
+                    LazyInstrumentationLoader.Add(new MySqlDataInitializer());
```

Comment:
I think we should reuse the same mechanism for ASP.NET Core (as a separate PR) (as originally done by @RassK).
It should simplify the codebase.

Reply:
#951